### ISSUE TITLE
[pickers] [a11y] Clock does not respond to Enter or space

### ISSUE THREAD

**Issue by pesvieira** [mui/mui-x#9636]
### Duplicates

- [X] I have searched the existing issues

### Latest version

- [X] I have tested the latest version

### Steps to reproduce 🕹

Link to live example: https://drive.google.com/file/d/1Qo4t07HgEPQbE7S5liTdARqaSxC77rRG/view?usp=sharing

### Current behavior 😯

Date Time Picker is not working when trying to select the time using enter or space keys. Also, there is a problem with navigating the minutes clock after choosing the time with the mouse.

**Comment by alexfauquette** [comment]
I confirm the clock still has those keyboard interaction issues. Can be tested [here](https://mui.com/x/react-date-pickers/date-time-picker/#available-components)

I would also add that the arrow key does not work on the minute's view.

However, the clock is not made to be used with a keyboard. We now have the [digital clock](https://mui.com/x/react-date-pickers/digital-clock/) which is much more a11y friendly
